tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debugger_selectors_test.ts

```diff
@@ -104,6 +207,84 @@ describe('debugger selectors', () => {
     });
   });
 
+  describe('getAlertTypesOfVisibleExecutionDigests', () => {
+    it('returns all-null array when there is no focused alert type', () => {
+      const state = createState(
+        createDebuggerState({
+          activeRunId: '__default_debugger_run__',
+          executions: {
+            numExecutionsLoaded: {
+              state: DataLoadState.LOADING,
+              lastLoadedTimeInMs: null,
+            },
+            executionDigestsLoaded: {
+              state: DataLoadState.NOT_LOADED,
+              lastLoadedTimeInMs: null,
+              pageLoadedSizes: {},
+              numExecutions: 0,
+            },
+            pageSize: 1000,
+            displayCount: 3,
+            focusIndex: null,
+            scrollBeginIndex: 0,
+            executionDigests: {
+              0: createTestExecutionDigest(),
+              1: createTestExecutionDigest(),
+              2: createTestExecutionDigest(),
+            },
+            executionData: {},
+          },
+        })
+      );
+
+      const alertTypes = getFocusAlertTypesOfVisibleExecutionDigests(state);
+      expect(alertTypes).toEqual([null, null, null]);
+    });
+
+    it('returns correct non-null array when there is focused alert type', () => {
+      const state = createState(
+        createDebuggerState({
+          activeRunId: '__default_debugger_run__',
+          alerts: createAlertsState({
+            focusType: AlertType.INF_NAN_ALERT,
+            executionIndices: {
+              [AlertType.INF_NAN_ALERT]: [0, 2, 3],
+            },
+          }),
+          executions: {
+            numExecutionsLoaded: {
+              state: DataLoadState.LOADING,
+              lastLoadedTimeInMs: null,
+            },
+            executionDigestsLoaded: {
+              state: DataLoadState.NOT_LOADED,
+              lastLoadedTimeInMs: null,
+              pageLoadedSizes: {},
+              numExecutions: 0,
+            },
+            pageSize: 1000,
+            displayCount: 3,
+            focusIndex: null,
+            scrollBeginIndex: 0,
+            executionDigests: {
+              0: createTestExecutionDigest(),
+              1: createTestExecutionDigest(),
+              2: createTestExecutionDigest(),
```

Comment:
I see that the indices are [0, 2, 3]. Is it okay not to have an entry for `3`?

Reply:
Even if we add an entry for 3, it'll be outside the scrolling window and hence invisible, due to `displayCount: 3`. But to humor that case, I added 3 here anyway. It doesn't change the assertions.